### ISSUE TITLE
Supporting secure websockets

### ISSUE THREAD

**Issue by chelliwell** [sahb1239/SAHB.GraphQLClient#68]
An essential requirement of my system is to use certificate-based TLS security between client and server.
I've been able to achieve this on queries by adding the certificates to a HttpClientHandler, using that to instantiate a HttpClient, then using the overridden GraphQLHttpClient.Default().
However I can't see a similar route for GraphQLSubscriptionWebSocketClient.Default(). One idea is to expose the ClientWebSocket, or at least the Options, from the GraphQLSubscriptionWebSocketClient class. To allow for an equivalent to
```
        public GraphQLSubscriptionWebSocketClient(IGraphQLFieldBuilder fieldBuilder, IGraphQLQueryGeneratorFromFields queryGenerator, IGraphQLDeserialization deserialization, CancellationToken cancellationToken)
        {
            _cancellationToken = cancellationToken;
            _webSocket.Options.AddSubProtocol("graphql-ws");

            X509CertificateCollection certCollection = new X509CertificateCollection
            {    new X509Certificate2(pfxFile, pfxPassword));   };
            _webSocket.Options.ClientCertificates = certCollection;

            this.fieldBuilder = fieldBuilder;
            this.queryGenerator = queryGenerator;
            this.deserialization = deserialization;
        }
```
_Obviously that itself is the wrong way to do it lol. I just wanted to illustrate the aim, and I've also used this code to check that the basic principle is correct - which it seems to be since it does a wss connection to my server._

I'd be interested to hear your thoughts. Note that I am no expert C# programmer, so there may be a better way to achieve it than simply exposing the ClientWebSocket.

**Comment by sahb1239** [comment]
Hi, Think the best way would be to support a new static Default method on GraphQLSubscriptionWebSocketClient which support passing the ClientWebSocket as a parameter. I think this is what corresponds closest to the GraphQLHttpClient default method which support passing in a HttpClient.

Thanks for your suggestion :)

**Comment by chelliwell** [comment]
I'll take a look at that route, and see if I can put together something acceptable. Will fire in a Pull Request if I get summat sorted.

**Comment by chelliwell** [comment]
> You could create a new static Default method which could call a new constructor which takes the websocket as a input parameter. I think that would be a good way to do it.

Ah, an additional GraphQLSubscriptionWebSocketClient constructor, coupled with an additional static IGraphQLSubscriptionWebSocketClient Default() ?

**Comment by sahb1239** [comment]
Yep that would solve the problem assigning to the readonly variable

**Comment by sahb1239** [comment]
This should now be fixed by #72
